bivrost-solidity-types/src/main/kotlin/pm/gnosis/model/SolidityBase.kt

```diff
@@ -26,10 +29,10 @@ object SolidityBase {
 
     interface DynamicType : Type {
         data class Parts(val static: String, val dynamic: String)
-
-        fun encodeParts(): Parts
     }
 
+    abstract class Collection<T : Type>(val hasDynamicItems: Boolean = false) : Type
```

Comment:
Not sure if we need the generic information here since we are not passing it through the super type

Reply:
I just wanted to force the subclasses to have that info, we could pull items all the way up here, right?